src/access_nri_intake/source/builders.py

```diff
@@ -378,9 +397,12 @@ def __init__(self, path):
         super().__init__(**kwargs)
 
     @classmethod
-    def parser(cls, file):
+    def parser(cls, file) -> dict:
         try:
-            match_groups = re.match(r".*/output\d+/([^/]*)/.*\.nc", file).groups()
+            # mypy gets upset as match can return None. I assume this is why we
+            # have try/except block in the first place? If so, we might be able
+            # to make this more explicit?
+            match_groups = re.match(r".*/output\d+/([^/]*)/.*\.nc", file).groups()  # type: ignore
```

Comment:
It looks to me like that idea is that we're just palming off any error as 'invalid asset', which is why the `except` clause catches basically everything.

Reply:
Yeah - I'm generally in favour of making try/except blocks as tight as possible, but if we can potentially have issues from the parser too lets leave this as is for now. Just pushed an update to this block.